src/fs.rs

```diff
@@ -387,16 +403,40 @@ impl<IO: Read + Write + Seek, TP, OCC> FileSystem<IO, TP, OCC> {
             FsInfoSector::default()
         };
 
+        let mut bpb_status_flags = bpb.status_flags();
+
         // if dirty flag is set completly ignore free_cluster_count in FSInfo
-        if bpb.status_flags().dirty {
+        if bpb_status_flags.dirty() {
+            if options.ignore_dirty_flag {
+                warn!("BPB is dirty, clearing dirty flag.");
+                bpb_status_flags.dirty = false;
+                write_bpb_status_flags(&mut disk, fat_type, bpb_status_flags)?;
+                bpb.set_status_flags(bpb_status_flags);
+            } else {
+                return Err(Error::DirtyFileSystem);
+            }
+
             fs_info.free_cluster_count = None;
         }
 
         // Validate the numbers stored in the free_cluster_count and next_free_cluster are within bounds for volume
         fs_info.validate_and_fix(total_clusters);
 
-        // return FileSystem struct
-        let status_flags = bpb.status_flags();
+        let mut fat_status_flags = read_fat_flags(&mut fat_slice::<&mut IO, IO::Error, IO>(&mut disk, &bpb), fat_type)?;
+        if fat_status_flags.dirty() {
+            if options.ignore_dirty_flag {
+                warn!("FAT is dirty, clearing dirty flag.");
```

Comment:
it's not BPB being dirty, it's entire filesystem. Please change it, e.g. "Filesystem is dirty (FAT), clearing..."